### PR TITLE
Overhaul of Section 5.1.1.

### PR DESCRIPTION
**Pull request by henkbirkholz**
* Renamed to: Structure and Content of Signed Statements
* Updated content based on this thread: https://mailarchive.ietf.org/arch/browse/scitt/?gbt=1&index=ON3uw6WDeC8o62bPgSt8RWNWg9M
* Added `x5t` and made both CWT_Claims `iss` and `sub` mandatory as per the output of discussions
* added refs and figure label (which are missing all over the place)

### CODE REVIEW

File: draft-ietf-scitt-architecture.md [hunk 1]
```diff
@@ -363,36 +366,41 @@ This section describes at a high level, the three main roles and associated proc
...
 
-Issuers MAY rotate verification keys at any time, or at a consistent cryptoperiod.
-Issuers MAY migrate to new signing and verification algorithms, but the Transparency Service remains responsible for admitting signed statements that match its policies.
+An Issuer identifier is required to be included in the COSE envelope. The protected header of a Signed Statement MUST include the `CWT Claims` header parameter as specified in {{Section 2 of CWT_CLAIMS_COSE}}. The CBOR map that constitutes the corresponding `CWT Claims` value MUST include the `Issuer Claim` (Claim label 1, see {{IANA.cwt}}) where the value MUST represent the Issuer identifier.
 
-The Issuer's identifier is required and appears in the `1 iss` claim of the `15 CWT_Claims` protected header of the Signed Statements' Envelope.
-The version of the key used to sign the Signed Statement is written in the `4 kid` protected header.
+An Subject identifier is also required to be included in the COSE envelope. The CBOR map that constitutes the corresponding `CWT Claims` value MUST include the `Subject Clain` (Claim label 2, see {{IANA.cwt}}) where the value MUST represent the Subject identifier.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:384]:
```suggestion
An Subject identifier is also required to be included in the COSE envelope. The CBOR map that constitutes the corresponding `CWT Claims` value MUST include the `Subject Claim` (Claim label 2, see {{IANA.cwt}}) where the value MUST represent the Subject identifier.
```

File: draft-ietf-scitt-architecture.md [hunk 2]
```diff
@@ -363,36 +366,41 @@ This section describes at a high level, the three main roles and associated proc
...
 Protected_Header = {
-  1   => int             ; algorithm identifier,
-  4   => bstr            ; Key ID (kid),
-  15  => CWT_Claims      ; CBOR Web Token Claims,
-  3   => tstr            ; payload type
+    1 => int             ; algorithm identifier
+    3 => tstr            ; payload type
+  ? 4 => bstr            ; Key ID (kid)
+   15 => CWT_Claims      ; CBOR Web Token Claims
+   34 => COSE_CertHash   ; x5t, hash of an X.509 certificate
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:400]: are additional properties allowed here? I think they should be.

File: draft-ietf-scitt-architecture.md [hunk 3]
```diff
@@ -363,36 +366,41 @@ This section describes at a high level, the three main roles and associated proc
 
 ## Signed Statement Issuance and Registration
 
-### Issuer Identity
+### Structure and Content of Signed Statements
+
+Before an Issuer is able to produce Signed Statements, it must first create an identifier and obtain an identity document that is acceptable to the Transparency Service.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:371]: Do we want to make registering identity documents mandatory prior to accepting signed statements from issuers?

This makes the replay verification simple, find the public key in the log, then find each signature.

File: draft-ietf-scitt-architecture.md [hunk 4]
```diff
@@ -363,36 +366,41 @@ This section describes at a high level, the three main roles and associated proc
...
+Transparency Services MUST support the capability to associate an X.509v3 certificate with a Signed Statement using a hash (thumbprint) of the certificate as specified in {{Section 2 of RFC9360}} by supporting the `x5t` COSE header parameter. The `x5t` COSE header parameter MUST be included in the protected header of a Signed Statement's COSE envelope. The mechanisms how Transparency Services obtain corresponding X.509v3 certificates, e.g., as part of enforcing Registration Policy, is out-of-scope of this document.
+Transparency Services MAY support many other identifier formats for identifying many other identity document formats. Alternative identifiers for identity documents MUST also be included in the protected header of the COSE envelope. Only one type of identity document identifier MUST be included in a Signed Statement's COSE envelope.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:373]:
```suggestion
Transparency Services MAY support other identifier formats for identifying other identity document formats. Alternative identifiers for identity documents MUST also be included in the protected header of the COSE envelope.
```

File: draft-ietf-scitt-architecture.md (hunk 2 above)
Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:400]: they were not allowed previously. While that does not make a lot of sense, I kept it that way for now.

File: draft-ietf-scitt-architecture.md (hunk 3 above)
Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:371]: I would very, very much like that. They'll be used in Registration Policies and those also should be registered. So - yeah +1 (to automation, resilience, and transparency!)

File: draft-ietf-scitt-architecture.md (hunk 4 above)
Comment (OR13) [reply draft-ietf-scitt-architecture.md:373]: reasoning being that `iss` and `kid` are alternative names for `x5t` so logically, if `iss` and `x5t` are present, we are allowing / requiring multiple identifiers.

Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:373]: So including several (potential conflicting identity document identifier is fine? Or do we want text, if there are multiple they MUST relate to the same identity document?

File: draft-ietf-scitt-architecture.md [hunk 5]
```diff
@@ -363,36 +366,41 @@ This section describes at a high level, the three main roles and associated proc
...
 
-Before an Issuer is able to produce Signed Statements, it must first create an identifier and obtain an identity document, that is acceptable to the Transparency Service.
-Transparency Services MAY support many different identity document formats.
+Issuers SHOULD use consistent identifiers for all their Statements about Artifacts to simplify authorization by Verifiers and auditing.
+If an Issuer uses multiple identifiers across their Statements, they MUST ensure that Statements signed under each identifier are consistent.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:376]:
```suggestion
```

Comment (OR13) [reply draft-ietf-scitt-architecture.md:376]: Let people reflect the graphs that exist today... even if they are not great. Policy can be used to adjust this, we don't need normative guidance here.

Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:376]: I am happy whenever we can stay less prescriptive. okay

File: draft-ietf-scitt-architecture.md (hunk 4 above)
Comment (OR13) [reply draft-ietf-scitt-architecture.md:373]: If my registration policy accepts your message, I see no reason for us to say more.

The industry pressure for multiple receipts will cause the alignment here, and I doubt we can normatively capture all the possible ways to indirectly refer to an "identity document".

File: draft-ietf-scitt-architecture.md [hunk 6]
```diff
@@ -359,40 +362,40 @@ The SCITT architecture consists of a very loose federation of Transparency Servi
...
+Transparency Services MAY support many other identifier formats for identifying many other identity document formats. Alternative identifiers for identity documents MUST also be included in the protected header of the COSE envelope. Only one type of identity document identifier MUST be included in a Signed Statement's COSE envelope.
 
-Issuers MAY rotate verification keys at any time, or at a consistent cryptoperiod.
-Issuers MAY migrate to new signing and verification algorithms, but the Transparency Service remains responsible for admitting signed statements that match its policies.
+As Issuers MUST rotate verification keys for signature checking in well-defined cryptoperiods (see {{KEY-MANAGEMENT}}).
+Issuers MAY migrate to new signing and verification algorithms, but the Transparency Service remains responsible for admitting Signed Statements that complies with its active Registration Policies.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:376]:
```suggestion
Transparency Services MUST specify their supported signature algorithms in their registration policies.
```

File: draft-ietf-scitt-architecture.md [hunk 7]
```diff
@@ -359,40 +362,40 @@ The SCITT architecture consists of a very loose federation of Transparency Servi
...
+The version of the key used to sign the Signed Statement MAY be included via the `kid` COSE header parameter in the protected header. Key discovery protocols are out-of-scope of this document.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:377]:
```suggestion
The `kid` header parameter MUST be present when the `x5t` header parameter is not present. Key discovery protocols are out-of-scope of this document.
```

File: draft-ietf-scitt-architecture.md [hunk 8]
```diff
@@ -359,40 +362,40 @@ The SCITT architecture consists of a very loose federation of Transparency Servi
...
 
-### Issuer Identity
-
-Before an Issuer is able to produce Signed Statements, it must first create an identifier and obtain an identity document, that is acceptable to the Transparency Service.
-Transparency Services MAY support many different identity document formats.
+### Structure and Content of Signed Statements
 
-Issuers SHOULD use consistent identifiers for all their Statements about Artifacts, to simplify authorization by Verifiers and auditing.
-If an Issuer uses multiple identifiers, they MUST ensure that statements signed under each identifier are consistent.
+Before an Issuer is able to produce Signed Statements, it must first create an identifier and obtain an identity document that is acceptable to the Transparency Service.
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:371]: A bit of tweaking as an Issuer can produce signed statements, which they may submit to any number of transparency services. The registration policy may fail if it doesn't support a type of identity, but as worded it sounds like you can't sign a statement, as opposed to the TS doesn't have to accept it.
I was trying to find an alternative way to say this, but is this sentence needed at all?
```suggestion
```

File: draft-ietf-scitt-architecture.md [hunk 9]
```diff
@@ -359,40 +362,40 @@ The SCITT architecture consists of a very loose federation of Transparency Servi
...
+Transparency Services MUST support the capability to associate an X.509v3 certificate with a Signed Statement using a hash (thumbprint) of the certificate as specified in {{Section 2 of RFC9360}} by supporting the `x5t` COSE header parameter. The `x5t` COSE header parameter MUST be included in the protected header of a Signed Statement's COSE envelope. The mechanisms how Transparency Services obtain corresponding X.509v3 certificates, e.g., as part of enforcing Registration Policy, is out-of-scope of this document.
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:372]: added line wrapping for PR feedback
```suggestion
Transparency Services MUST support the capability to associate an X.509v3 certificate with a Signed Statement using a hash (thumbprint) of the certificate as specified in {{Section 2 of RFC9360}} by supporting the `x5t` COSE header parameter.
The `x5t` COSE header parameter MUST be included in the protected header of a Signed Statement's COSE envelope.
The mechanisms for how Transparency Services obtain corresponding X.509v3 certificates, e.g., as part of enforcing a Registration Policy, is out-of-scope of this document.
```

File: draft-ietf-scitt-architecture.md [hunk 10]
```diff
@@ -359,40 +362,40 @@ The SCITT architecture consists of a very loose federation of Transparency Servi
...
+Transparency Services MAY support many other identifier formats for identifying many other identity document formats. Alternative identifiers for identity documents MUST also be included in the protected header of the COSE envelope. Only one type of identity document identifier MUST be included in a Signed Statement's COSE envelope.
 
-Issuers MAY rotate verification keys at any time, or at a consistent cryptoperiod.
-Issuers MAY migrate to new signing and verification algorithms, but the Transparency Service remains responsible for admitting signed statements that match its policies.
+As Issuers MUST rotate verification keys for signature checking in well-defined cryptoperiods (see {{KEY-MANAGEMENT}}).
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:375]:
```suggestion
Issuers MUST rotate verification keys for signature checking in well-defined cryptoperiods (see {{KEY-MANAGEMENT}}).
```

File: draft-ietf-scitt-architecture.md (hunk 6 above)
Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md:376]: @OR13, doesn't this pull in the standardization of registration policies, if we say they must specify their supported algorithms?

File: draft-ietf-scitt-architecture.md [hunk 11]
```diff
@@ -359,40 +362,40 @@ The SCITT architecture consists of a very loose federation of Transparency Servi
...
+Issuers MAY migrate to new signing and verification algorithms, but the Transparency Service remains responsible for admitting Signed Statements that complies with its active Registration Policies.
+The version of the key used to sign the Signed Statement MAY be included via the `kid` COSE header parameter in the protected header. Key discovery protocols are out-of-scope of this document.
 
-The Issuer's identifier is required and appears in the `1 iss` claim of the `15 CWT_Claims` protected header of the Signed Statements' Envelope.
-The version of the key used to sign the Signed Statement is written in the `4 kid` protected header.
+An Issuer identifier and a Subject identifier are required to be included in the COSE envelope. Consequently, the protected header of a Signed Statement MUST include the `CWT Claims` header parameter as specified in {{Section 2 of CWT_CLAIMS_COSE}}. The CBOR map that constitutes the corresponding `CWT Claims` value MUST include the `Issuer Claim` (Claim label 1) and the `Subject Claim` (Claim label 2) {{IANA.cwt}}.
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:379]: Line formatting for PR feedback and consistency
```suggestion
An Issuer identifier and a Subject identifier are required to be included in the COSE envelope. Consequently, the protected header of a Signed Statement MUST include the `CWT Claims` header parameter as specified in {{Section 2 of CWT_CLAIMS_COSE}}.
The CBOR map that constitutes the corresponding `CWT Claims` value MUST include the `Issuer Claim` (Claim label 1) and the `Subject Claim` (Claim label 2) {{IANA.cwt}}.
```

**Review by SteveLasker — COMMENTED** [review]
A few nits for formatting and a question regarding how detailed we need/should be about signed statements.

File: draft-ietf-scitt-architecture.md (hunk 7 above)
Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md:379]: nit: line wrap sentences, please.

File: draft-ietf-scitt-architecture.md [hunk 12]
```diff
@@ -359,40 +362,43 @@ The SCITT architecture consists of a very loose federation of Transparency Servi
...
-
-Before an Issuer is able to produce Signed Statements, it must first create an identifier and obtain an identity document, that is acceptable to the Transparency Service.
-Transparency Services MAY support many different identity document formats.
+### Structure and Content of Signed Statements
 
-Issuers SHOULD use consistent identifiers for all their Statements about Artifacts, to simplify authorization by Verifiers and auditing.
-If an Issuer uses multiple identifiers, they MUST ensure that statements signed under each identifier are consistent.
+Before an Issuer is able to produce Signed Statements, it must first create an identifier and obtain an identity document that is acceptable to the Transparency Service.
+Transparency Services MUST support the capability to associate an X.509v3 certificate with a Signed Statement using a hash (thumbprint) of the certificate as specified in {{Section 2 of RFC9360}} by supporting the `x5t` COSE header parameter.
+The `x5t` COSE header parameter MUST be included in the protected header of a Signed Statement's COSE envelope.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:373]:
```suggestion
The `x5t` COSE header parameter MUST be included in the protected header of a Signed Statement's COSE envelope.
When `x5t` is present, `iss` MUST be a string with a value between 1 and 8192 characters in length that fits the regular expression of a distinguished name.
```

Comment (CBonnell) [reply draft-ietf-scitt-architecture.md:373]: I suggest specifying the second sentence (on line 374) as:
```When `x5t` is present, `iss` MUST be the {{!RFC 4514}} string representation of the subject distinguished name of the certificate whose thumbprint is included in `x5t`.
```

File: draft-ietf-scitt-architecture.md (hunk 8 above)
Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:371]: With the "new" title and structure I felt that this lead-in sentence now felt off. Less, crisp text is better. Any objections?

File: draft-ietf-scitt-architecture.md (hunk 12 above)
Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:373]: :+1:

File: draft-ietf-scitt-architecture.md (hunk 4 above)
Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md]: Good. I agree. Maybe some Security Considerations? Created #170 as a reminder.

File: draft-ietf-scitt-architecture.md (hunk 2 above)
Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:399]: 17h later, it still makes no sense, changed it via suggestion feature.

File: draft-ietf-scitt-architecture.md [hunk 13]
```diff
@@ -359,40 +362,44 @@ The SCITT architecture consists of a very loose federation of Transparency Servi
...
-  15  => CWT_Claims      ; CBOR Web Token Claims,
-  3   => tstr            ; payload type
+    1 => int             ; algorithm identifier
+    3 => tstr            ; payload type
+  ? 4 => bstr            ; Key ID (kid)
+   15 => CWT_Claims      ; CBOR Web Token Claims
+   34 => COSE_CertHash   ; x5t, hash of an X.509 certificate
 }
 ~~~
+{: #fig-signed-statement title="CDDL definition for SCITT Signed Statements"}
```
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:402]:
```suggestion
   34 => COSE_CertHash   ; x5t, hash of an X.509 certificate
   * int => any
}
~~~
{: #fig-signed-statement title="CDDL definition for SCITT Signed Statements"}
```

**Review by henkbirkholz — COMMENTED** [review]
Addressed some feedback and added a suggestion

File: draft-ietf-scitt-architecture.md (hunk 7 above)
Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:380]: If there are no objections, this suggestion is good to go.

File: draft-ietf-scitt-architecture.md (hunk 6 above)
Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:378]: Last open item for discussion today

Comment (OR13) [reply draft-ietf-scitt-architecture.md:378]: its ok to say MUST and leave it as an implementation detail

**Review by aj-stein-nist — APPROVED** [review]

**Review by OR13 — APPROVED** [review]

**Review by SteveLasker — APPROVED** [review]
LGTM

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM

**Pull request merged**